wrapper.nix

```diff
@@ -0,0 +1,29 @@
+{ lib, writeScriptBin
+, hie80, hie82, hie84
+}:
+
+writeScriptBin "hie" ''
+#!/usr/bin/env bash
+
+GHCBIN='ghc'
+if type -f stack >/dev/null; then
+  # Prefer stack, if available
+  GHCBIN='stack ghc --'
```

Comment:
Not sure this works well on NixOS:

```
$ stack ghc -- --version
error: attribute 'ghc801' missing, at (string):1:48
(use '--show-trace' to show detailed location information)

$ stack --version|head -n 1
Version 1.7.1 x86_64
```

Reply:
Oh! It works inside a Haskell project, but not outside. I will have to go upstream to find why.

Reply:
The problem seems to be that Stack [hardcodes](https://github.com/commercialhaskell/stack/blob/6fd731ead944df0479af1a60bdbbc88eaada775b/src/Stack/Config/Nix.hs#L56) compiler versions because Nixpkgs does not provide a link to the latest in each branch.